test/arch/aarch64/unit/asm_test.py

```diff
@@ -0,0 +1,65 @@
+#! /usr/bin/env python
+import sys
+import os
+
+from miasm2.core.cpu import parse_ast
+from miasm2.arch.aarch64.arch import mn_aarch64, base_expr, variable
+from miasm2.core import parse_asm
+from miasm2.expression.expression import *
+from miasm2.core import asmbloc
+from elfesteem.strpatchwork import StrPatchwork
+from miasm2.analysis.machine import Machine
+from miasm2.jitter.csts import *
+from pdb import pm
+
+
+filename = os.environ.get('PYTHONSTARTUP')
+if filename and os.path.isfile(filename):
+    execfile(filename)
+
+
+reg_and_id = dict(mn_aarch64.regs.all_regs_ids_byname)
+
+class Asm_Test(object):
+    def __init__(self):
+        self.myjit = Machine("aarch64l").jitter()
+        self.myjit.init_stack()
+
+        self.myjit.jit.log_regs = False
+        self.myjit.jit.log_mn = False
+
+
+    def __call__(self):
+        self.asm()
+        self.run()
+        self.check()
+
+
+    def asm(self):
+        blocs, symbol_pool = parse_asm.parse_txt(mn_aarch64, 'l', self.TXT,
+                                                 symbol_pool = self.myjit.ir_arch.symbol_pool)
+        # fix shellcode addr
+        symbol_pool.set_offset(symbol_pool.getby_name("main"), 0x0)
+        s = StrPatchwork()
+        patches = asmbloc.asm_resolve_final(mn_aarch64, blocs[0], symbol_pool)
+        for offset, raw in patches.items():
+            s[offset] = raw
+
+        s = str(s)
```

Comment:
Yep, but the intermediate `s` var is useless, the code is really understandable without and it avoid an indirection